src/api/bonding-curve.js

```diff
@@ -0,0 +1,176 @@
+  totalETH = toBN(totalETH)
+  totalPNK = toBN(totalPNK)
+  spread = toBN(spread)
+
+  return ETH.mul(totalPNK)
+    .mul(SPREAD_DIVISOR)
+    .div(totalETH.add(ETH))
+    .div(SPREAD_DIVISOR.add(spread))
+    .toString()
+}
+
+/** Given an input PNK amount and the state values of the bonding curve contract,
+ *  compute the amount of ETH that the PNK is sold for using the same formula as
+ *  the bonding curve contract. Note this duplicates the algorithm from the contract
+ *  but we can't call the contract because that way the turnaround would be too
+ *  slow for a responsive UI.
+ *  @param {string} inputPNK User input PNK amount. May not be a valid number.
+ *  @param {BigNumber} totalETH 'totalETH' value of the contract.
+ *  @param {BigNumber} totalPNK 'totalPNK' value of the contract.
+ *  @param {BigNumber} spread 'spread' value of the contract.
+ *  @returns {string} Amount of ETH in wei.
+ */
+export function estimateETH(inputPNK, totalETH, totalPNK, spread) {
+  var PNK
+  try {
+    PNK = decimalStringToWeiBN(inputPNK)
+  } catch (_) {
+    return '0'
+  }
+  // convert all to BN from BigNumber
+  totalETH = toBN(totalETH)
+  totalPNK = toBN(totalPNK)
+  spread = toBN(spread)
+
+  return totalETH
+    .mul(PNK)
+    .mul(SPREAD_DIVISOR)
+    .div(totalPNK.add(PNK))
+    .div(SPREAD_DIVISOR.add(spread))
+    .toString()
+}
+
+/**
+ * Provides interaction with the bonding curve contract on the blockchain.
+ */
+export default class BondingCurve extends ContractImplementation {
+  /**
+   * Create new Bonding Curve Implementation.
+   * @param {object} web3Provider - web3 instance.
+   * @param {string} contractAddress - Address of the Bonding Curve contract.
+   */
+  constructor(web3Provider, contractAddress) {
+    super(web3Provider, bondingCurveArtifact, contractAddress)
+  }
+
+  /**
+   * Fetch the total amount of ETH in the bonding curve.
+   * @returns {number} - The total amount of ETH as a BigNumber.
+   */
+  getTotalETH = async () => {
+    await this.loadContract()
+    try {
+      return this.contractInstance.totalETH()
+    } catch (err) {
+      console.error(err)
+      throw new Error('Unable to fetch totalETH from the bonding curve')
+    }
+  }
+
+  /**
+   * Fetch the total amount of bonded token in the bonding curve.
+   * @returns {number} - The total amount of bonded token as a BigNumber.
+   */
+  getTotalTKN = async () => {
+    await this.loadContract()
+    try {
+      return this.contractInstance.totalTKN()
+    } catch (err) {
+      console.error(err)
+      throw new Error('Unable to fetch totalTKN from the bonding curve')
+    }
+  }
+
+  /**
+   * Fetch the spead of the bonding curve.
+   * @returns {number} - The spread as a BigNumber.
+   */
+  getSpread = async () => {
+    await this.loadContract()
+    try {
+      return this.contractInstance.spread()
+    } catch (err) {
+      console.error(err)
+      throw new Error('Unable to fetch spread from the bonding curve')
```

Comment:
This file is unnecessary and does not follow the rest of the codebase, you can call contract methods directly from sagas.